src/main/java/org/minions/devfund/abel/Squarelotron.java

```diff
@@ -0,0 +1,184 @@
+package org.minions.devfund.abel;
+
+import java.util.Arrays;
+
+/**
+ * Class Squarelotron consists basically of a matrix of numbers.
+ */
+public class Squarelotron {
+
+
+    private int[][] squarelotronMatrix;
+    private int size;
+
+    /**
+     * Constructor to fill a normal Squarelotron.
+     *
+     * @param size size of the Squarelotron matrix.
+     */
+    public Squarelotron(int size) {
+        squarelotronMatrix = new int[size][size];
+        int number = 1;
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                squarelotronMatrix[i][j] = number;
+                number++;
+            }
+        }
+        this.size = size;
+    }
+
+    /**
+     * Constructor to assign values to the Squarelotron.
+     *
+     * @param squarelotron value assigned to the Squarelotron matrix.
+     */
+    public Squarelotron(final int[][] squarelotron) {
+        this.squarelotronMatrix = deepCopy(squarelotron);
+        this.size = this.squarelotronMatrix.length;
+    }
+
+    /**
+     * Get the Squarelotron.
+     *
+     * @return The Squarelotron.
+     */
+    public int[][] getSquarelotronMatrix() {
+        return deepCopy(squarelotronMatrix);
+    }
+
+    /**
+     * This method performs the Upside-Down Flip of the squarelotronMatrix, as described above,
+     * and returns the new squarelotronMatrix.
+     *
+     * @param ring The ring to be modified.
+     * @return A new Squarelotron object.
+     */
+    public Squarelotron upsideDownFlip(int ring) {
+        Function<Integer, Integer, Integer> upsideDown = (i, j) -> squarelotronMatrix[size - i - 1][j];
+        return new Squarelotron(flip(ring, upsideDown));
+    }
+
+    /**
+     * This method performs the Main Diagonal Flip of the squarelotronMatrix, as described above,
+     * and returns the new squarelotronMatrix.
+     *
+     * @param ring The ring to be modified.
+     * @return A new Squarelotron object.
+     */
+    public Squarelotron mainDiagonalFlip(int ring) {
+        Function<Integer, Integer, Integer> upsideDown = (i, j) -> squarelotronMatrix[j][i];
+        return new Squarelotron(flip(ring, upsideDown));
+    }
+
+    /**
+     * Method to flip based on a Function.
+     *
+     * @param ring The ring to be modified.
+     * @param fn   The lambda function to execute.
+     * @return 2d array flipped.
+     */
+    private int[][] flip(int ring, final Function<Integer, Integer, Integer> fn) {
+        int[][] copy = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (isInRing(ring, i, j)) {
```

Comment:
Instead of using an if and else conditional, you can use a ternary operator 
```java
copy[i][j] = isInRing(ring, i, j) ? fn.apply(i, j) : squarelotronMatrix[i][j];
```